common/raw_hashtable_metadata_group.h

```diff
@@ -830,19 +933,22 @@ inline auto MetadataGroup::SIMDClearDeleted() -> void {
 #endif
 }
 
-inline auto MetadataGroup::SIMDMatch(uint8_t present_byte) const -> MatchRange {
+inline auto MetadataGroup::SIMDMatch(uint8_t tag) const -> MatchRange {
   MatchRange result;
 #if CARBON_NEON_SIMD_SUPPORT
-  auto match_byte_vec = vdup_n_u8(present_byte | PresentMask);
+  auto match_byte_vec = vdup_n_u8(tag | PresentMask);
   auto match_byte_cmp_vec = vceq_u8(byte_vec, match_byte_vec);
   uint64_t mask = vreinterpret_u64_u8(match_byte_cmp_vec)[0];
-  // Mask in scalar to fold with testing for zero.
+  // The mask in the range is likely to be tested for zero right after this
+  // routine, and that test can often be folded into an integer constant mask,
+  // so rather than doing the mask in SIMD before we extract the mask, we do it
+  // here to allow that combination.
```

Comment:
I find this comment confusing. Perhaps using the term "mask" too often with multiple different senses?

I also don't see any "mask" operation being performed on `result` here, which is what I'd expect given this comment.